### PR TITLE
Issue 503: Updated logic so next button enables/disables correctly

### LINKED ISSUE #503: NextButton disables before showing all the slides
`NextButton` gets disabled before the last slides are shown. This happens when `cellAlign` prop is set either to `center` or `right` and `slidesToScroll` prop value is more than 1.
Currently in the docs we mentioned that for this feature `cellAlign` `right` or `center` the user needs to set `slidesToScroll` to default( which is 1).

nukavideo

### PR DESCRIPTION
**Pull request by mariano-formidable**
### Description

This PR updates the logic for the enabling/disabling the next button so it works across (hopefully) all different configurations so `slideCount`, `slidesToShow`, `slidesToScroll`, and alignment.

Fixes #503 

#### Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

### How Has This Been Tested?

I manually tested multiple configurations and added a couple more spec/e2e tests.

### Screenshots

Normal slides:
normal-next

Partial Slides:
partial-next

Extreme example, slides to show is 6, and slides to scroll is 2:
slides-to-show-next

Extreme example, slides to show is 1.25, and slides to scroll is 1.25:
slides-to-scroll-next

### CODE REVIEW

File: test/specs/default-controls.test.js [hunk 1]
```diff
@@ -99,272 +99,66 @@ describe('<PagingDots />', () => {
 });
 
 describe('<NextButton />', () => {
-  describe('#nextButtonDisable for cellAlign = `left`', () => {
```
Comment (mariano-formidable) [inline test/specs/default-controls.test.js]: With the new logic, I had to re-write all the `nextButtonDisable` tests and I greatly simplified them.  Just checking for happy path basic logic.

File: test/e2e/carousel.test.js [hunk 2]
```diff
@@ -84,6 +110,38 @@ describe('Nuka Carousel', () => {
...
+
+        it('should allow you to navigate entire slide deck when center aligned', async () => {
+          await expect(page).toClick('button', {
+            text: 'Toggle Partially Visible Slides'
+          });
+          await expect(page).toClick('button', { text: 'Center' });
+          await page.waitFor(600); // need to let slide transition complete
+
+          await nextThroughAllSlides();
+        });
```
Comment (mariano-formidable) [inline test/e2e/carousel.test.js]: This test when the slides are `center` aligned specifically did _not_ work (the next button would disable too early) with the old logic, but now it works 😎

File: test/e2e/carousel.test.js [hunk 3]
```diff
@@ -75,6 +44,63 @@ describe('Nuka Carousel', () => {
 
       await expect(hiddenStyles.visibility).toMatch('hidden');
     });
+
+    describe('when in wrapAround mode', () => {
```
Comment (mariano-formidable) [inline test/e2e/carousel.test.js]: I moved these existing tests into a separate describe so each `it` statement didn't have to say "when wrapAround is true"

File: demo/app.js [hunk 4]
```diff
@@ -60,7 +60,7 @@ class App extends React.Component {
                   : 'none'
               }}
             >
-              Nuka Carousel: Slide {currentSlide + 1}
+              Nuka Carousel: Slide {Math.ceil(currentSlide) + 1}
```
Comment (mariano-formidable) [inline demo/app.js]: Depending how the slidesToShow is set, sometimes the current slide can come back as a fraction (not sure if this is a bug or a feature), so I'm rounding it to `ceil` so they always appear as a solid int for testing purposes.

File: README.md [hunk 5]
```diff
@@ -76,7 +76,7 @@ Or on CodeSandBox at the following url: <a href="https://codesandbox.io/s/04wxlo
...
+| cellAlign                  | `React.PropTypes.oneOf(['left', 'center', 'right'])`                                                                                                                                                                                                                   | When displaying more than one slide, sets which position to anchor the current slide to. **Is overridden to `left` when `transitionMode="fade"`**                                               |                                                                                                                    |
```
Comment (mariano-formidable) [inline README.md]: We no longer need to warn users about setting `slidesToScroll` to the default value, should work with whatever they type in :D

Comment (sarmeyer) [reply README.md]: Awesome!!

File: test/specs/default-controls.test.js (hunk 1 above)
Comment (sarmeyer) [reply test/specs/default-controls.test.js]: 👍

**Pull request merged**